lib/restHandlers.js

```diff
@@ -0,0 +1,226 @@
+(function() {
+	'use strict';
+
+	var _ = require('lodash'),
+		qs = require('qs'),
+		apiUrl = require(__dirname + '/apiUrl'),
+		log = require(__dirname + '/log'),
+		rest = require(__dirname + '/rest');
+
+	function RestHandlers(options) {
+		var defaults = {
+			apiToken: null
+		};
+
+		this.options = _.extend({}, defaults, options || {});
+
+		return this;
+	}
+
+	RestHandlers.prototype.genericResponse = function(method, object, responseBody, callback, rawRequest, rawResponse) {
+
+		if (_.isString(responseBody)) {
+			try {
+				responseBody = JSON.parse(responseBody);
+			}
+			catch (err) {}
+		}
+
+		log('Handling response of ' + method + ' ' + object + ': ' + rawRequest.url.href + (rawRequest.url.href.indexOf(rawRequest.url.query) === -1 ? '?' + rawRequest.url.query : ''));
+
+		if (responseBody.success === true) {
+			if (_.isFunction(callback)) {
+				callback(null, responseBody.data, responseBody.additional_data, rawRequest, rawResponse);
+			}
+		}
+		else {
+			if (rawResponse.statusCode && (rawResponse.statusCode.toString().substr(0,1) == '4' || rawResponse.statusCode.toString().substr(0,1) == '5')) {
```

Comment:
could be a bit neater. if you pull out the status code before you can shorten this if a lot. also, this if can be shifted left one as part of the previous if